app/Policies/UserPolicy.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Policies;
+
+use App\Models\User;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class UserPolicy
+{
+	use HandlesAuthorization;
+
+	public const ADMIN = 'admin';
+	public const UPLOAD = 'upload';
+	public const EDIT_SETTINGS = 'editSettings';
+
+	/**
+	 * Perform pre-authorization checks.
+	 *
+	 * @param \App\Models\User $user
+	 * @param string           $ability
+	 *
+	 * @return void|bool
+	 */
+	public function before(?User $user, $ability)
+	{
+		if ($this->admin($user)) {
+			return true;
+		}
+	}
+
+	/**
+	 * This defines if the user is admin.
+	 *
+	 * @param User|null $user
+	 *
+	 * @return bool
+	 */
+	public function admin(?User $user): bool
+	{
+		return $user?->id === 0;
+	}
```

Comment:
Why is the `$user` parameter nullable? The return value is `false` anyway in that case.

```suggestion
	public function admin(User $user): bool
	{
		return $user->id === 0;
	}
```

Reply:
because then I can directly use it in `before()`

Reply:
Resolve ?